lib/utils.js

```diff
@@ -10,7 +11,81 @@ const delay = function(ms) {
   return new Promise(resolve => setTimeout(() => resolve(), ms));
 };
 
+/**
+ * Checks if a provided address is actively listening for incoming connections
+ *
+ * @param {string} host the host to connect to
+ * @param {number} port the port to check for availability
+ * @param {function} callback
+ */
+const checkAvailable = (host, port, callback) => {
+  const socket = new net.Socket();
+  function cleanup() {
```

Comment:
I feel like we can make this a standalone `function cleanup(socket)` so we're not making and destroying functions on the fly.

Reply:
sure